sygnal/gcmpushkin.py

```diff
@@ -721,15 +728,28 @@ def _build_data(
         if n.prio == "low":
             data["prio"] = "normal"
 
+        counts: Dict[str, Any] = {}
         if send_badge_counts and getattr(n, "counts", None):
             if api_version is APIVersion.Legacy:
-                data["unread"] = n.counts.unread
-                data["missed_calls"] = n.counts.missed_calls
+                if n.counts.unread:
+                    counts["unread"] = n.counts.unread
+                if n.counts.missed_calls:
+                    counts["missed_calls"] = n.counts.missed_calls
             elif api_version is APIVersion.V1:
                 if n.counts.unread:
-                    data["unread"] = str(n.counts.unread)
+                    counts["unread"] = str(n.counts.unread)
                 if n.counts.missed_calls:
-                    data["missed_calls"] = str(n.counts.missed_calls)
+                    counts["missed_calls"] = str(n.counts.missed_calls)
+
+        if not data.get("room_id") and not data.get("event_id") and not counts:
+            logger.info(
```

Comment:
```suggestion
            logger.debug(
```

see https://github.com/element-hq/sygnal/pull/424#discussion_r2460934159